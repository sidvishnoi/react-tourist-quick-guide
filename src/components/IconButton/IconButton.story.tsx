import { action } from '@storybook/addon-actions';
import { color, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import * as React from 'react';
import IconButton from '.';

const stories = storiesOf('IconButton', module);
stories.addDecorator(withKnobs);

stories.add('with simple icon text', () => {
  return (
    <IconButton
      icon={text('text', 'âˆ†')}
      background={color('background', 'gold')}
      color={color('color', 'crimson')}
      altColor={color('altColor', '#fff')}
      title={text('title', 'TITLE')}
    />
  );
});

stories.add('with onClick handler', () => {
  return <IconButton icon="âˆ†" onClick={() => action('onClick')()} />;
});
