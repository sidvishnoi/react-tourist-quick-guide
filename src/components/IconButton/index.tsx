import * as React from 'react';
import styled from 'styled-components';

interface IconButtonStyleProps {
  background: string;
  color: string;
  altColor: string;
}

const IconButton = styled.button<IconButtonStyleProps>`
  background: ${props => props.background};
  border: none;
  padding: 0.5rem;
  margin: 0.1rem;
  color: ${props => props.color};

  :hover,
  :focus {
    background: ${props => props.color};
    color: ${props => props.altColor};
    cursor: pointer;
  }
`;

export interface IconButtonProps extends Partial<IconButtonStyleProps> {
  icon: string;
  onClick?: () => void;
  [prop: string]: any;
}

export default (props: IconButtonProps) => {
  const {
    icon,
    color = 'crimson',
    altColor = '#fff',
    background = 'rgba(255, 255, 255, 0.2)',
    onClick,
    ...rest
  } = props;
  return (
    <IconButton
      background={background}
      color={color}
      altColor={altColor}
      onClick={onClick}
      {...rest}
    >
      {props.icon}
    </IconButton>
  );
};
