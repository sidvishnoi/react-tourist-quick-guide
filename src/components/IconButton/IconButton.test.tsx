import { mount } from 'enzyme';
import * as Enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import React = require('react');
import * as renderer from 'react-test-renderer';
Enzyme.configure({ adapter: new Adapter() });

import IconButton from '.';

describe('Components - IconButton', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <IconButton icon="§" title="PASS" background="red" />,
    );
    expect(tree).toMatchSnapshot();
  });

  it('calls onClick', () => {
    const onClick = jest.fn();
    const output = mount(<IconButton icon="∆" onClick={onClick} />);
    expect(onClick).not.toHaveBeenCalled();
    output.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
