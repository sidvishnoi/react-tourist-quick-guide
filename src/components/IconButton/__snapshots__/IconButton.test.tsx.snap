// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - IconButton matches snapshot 1`] = `
.c0 {
  background: red;
  border: none;
  padding: 0.5rem;
  margin: 0.1rem;
  color: crimson;
}

.c0:hover,
.c0:focus {
  background: crimson;
  color: #fff;
  cursor: pointer;
}

<button
  className="c0"
  color="crimson"
  title="PASS"
>
  ยง
</button>
`;
